pkg/frontend/querymiddleware/instrumentation.go

```diff
@@ -15,6 +15,12 @@ import (
 	"github.com/prometheus/client_golang/prometheus/promauto"
 )
 
+type instrumentMiddleware struct {
```

Comment:
Note to reviewers: this is a small refactoring to make the instrument middleware detectable via reflection. Before it was an anonymous `HandlerFunc`.